packages/legacy-client/src/ontology-runtime/ontologyProvider/calls/listLinkedObjects.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ListLinkedObjectsError, OntologyObject, Result } from "../..";
+import {
+  handleListLinkedObjectsError,
+  ListLinkedObjectsErrorHandler,
+} from "../..";
+import type { ClientContext } from "./ClientContext";
+import { getLinkedObjectsPage } from "./getLinkedObjectsPage";
+import { wrapResult } from "./util/wrapResult";
+
+export function listLinkedObjects<T extends OntologyObject>(
+  context: ClientContext,
+  sourceApiName: string,
+  primaryKey: any,
+  linkTypeApiName: T["__apiName"],
+): Promise<Result<T[], ListLinkedObjectsError>> {
```

Comment:
Old style error handling still, but this breaks the listLinkedObjects function out of OntologyProvider.
Thoughts?

Reply:
Yeah this is much nicer than the big class!

Reply:
We can refactor the error handling later